client/app/scripts/superdesk-authoring/metadata/metadata.js

```diff
@@ -27,6 +27,22 @@ function MetadataCtrl($scope, desks, metadata, $filter, privileges, adminPublish
         });
     };
 
+    $scope.$watch('item.publish_schedule_date', function() {
```

Comment:
Thanks for those Petr, I'll be posting some bug fixes, I'll include these improvements as well.